gateway-ha/src/main/java/io/trino/gateway/ha/module/QueryCountBasedRouterProvider.java

```diff
@@ -13,22 +13,32 @@
  */
 package io.trino.gateway.ha.module;
 
+import com.google.inject.AbstractModule;
 import com.google.inject.Scopes;
+import io.airlift.log.Logger;
 import io.trino.gateway.ha.config.HaGatewayConfiguration;
 import io.trino.gateway.ha.router.QueryCountBasedRouter;
 import io.trino.gateway.ha.router.RoutingManager;
 
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+
 public class QueryCountBasedRouterProvider
-          extends RouterBaseModule
+        extends AbstractModule
 {
+    private static final Logger logger = Logger.get(QueryCountBasedRouterProvider.class);
+
     public QueryCountBasedRouterProvider(HaGatewayConfiguration configuration)
     {
-        super(configuration);
+        // no-op
```

Comment:
Why does the constructor still take `HaGatewayConfiguration` parameter?

Reply:
We require all modules to have this signature. Otherwise it can't be instantiated
https://github.com/trinodb/trino-gateway/blob/d718b3fc5f290bee8eec5530881f63cd61ea4f33/gateway-ha/src/main/java/io/trino/gateway/baseapp/BaseApp.java#L73-L74